### PR TITLE
Jgrgicevic/sw 727 resource list component

### PR DESCRIPTION
**Pull request by gjuro87**

### CODE REVIEW

**Comment by blfpd** [comment]
The table search doesn’t pickup the name. I’m not sure how long would it be to fix it.
Do you plan to add this feature in a later commit/MR?

File: frontend/plugins/resources/src/components/LanguagesPage/LanguagesPage.tsx [hunk 1]
```diff
@@ -1,12 +1,91 @@
...
-
-const LanguagesPage = () => (
-  <Page themeId="home">
-    <Content>
-        <div>Welcome to language page!</div>
-    </Content>
-  </Page>
-);
+import {Content, Page, Table, TableColumn, TableFilter} from '@backstage/core';
+import {DummyResponse, Edge, Resource} from "../../../../../packages/zoo-api";
```
Comment (blfpd) [inline frontend/plugins/resources/src/components/LanguagesPage/LanguagesPage.tsx]: Did you try with just `from "zoo-api";`? It should work as it is in `package.json`

File: frontend/plugins/resources/src/components/LanguagesPage/LanguagesPage.tsx [hunk 2]
```diff
@@ -1,12 +1,91 @@
...
+import ResourceLink from "../TableComponents/ResourceLink";
+import {Chip, Grid} from "@material-ui/core";
+import ResourceTypeLabel from "../TableComponents/ResourceTypeLabel";
+import Alert from "@material-ui/lab/Alert";
+
+
+const generateTableData: (resources: Array<Resource>) => Array<{}> = (resources) => {
+  const data: Array<{}> = [];
+  for(let i=0; i<resources.length; i++){
```
Comment (blfpd) [inline frontend/plugins/resources/src/components/LanguagesPage/LanguagesPage.tsx]: I’d prefer:
```js
for (const resource of resources) {
```

Ref: [for...of in MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of)

…but I guess you could do the same with `.map()`:

```js
return resources.map(resource => ({
	usage: resource.usageCourt,
	…
}))
```

File: frontend/plugins/resources/src/components/LanguagesPage/LanguagesPage.tsx [hunk 3]
```diff
@@ -1,12 +1,91 @@
...
+    if (response.error.message === "[Network] Unauthorized") {
+      return <Grid item>
+        <Alert severity="error">Unauthorized. Please check your Zoo token in the Settings.</Alert>
+      </Grid>;
+    }
+    return <Grid item>
+      <Alert severity="error">{response.error.message}</Alert>
+    </Grid>;
+  }
+  const resources: Resource[] = response.data.resources.edges.map((edge: Edge) => edge.node);
```
Comment (blfpd) [inline frontend/plugins/resources/src/components/LanguagesPage/LanguagesPage.tsx]: You could use my handy [`unwrap` function](https://github.com/kiwicom/the-zoo/blob/hangar/frontend/packages/zoo-api/src/client.ts#L32-L40) here:

```js
const resources = unwrap<Resource>(response.data.resources);
```

File: frontend/plugins/resources/src/components/LibrariesPage/LibrariesPage.tsx [hunk 4]
```diff
@@ -1,13 +1,108 @@
...
+    data.push({
+      usage: resources[i].usageCount,
+      name: <ResourceLink name={resources[i].name} id={resources[i].id} />,
+      version: <Chip label={resources[i].version} />,
+      type: <ResourceTypeLabel name={resources[i].type} />,
+      kind: <ResourceKind name={resources[i].name} />,
+    });
+  }
+
+  return data;
```
Comment (blfpd) [inline frontend/plugins/resources/src/components/LibrariesPage/LibrariesPage.tsx]: Same comment regarding `return resources.map()` :arrow_up:

File: frontend/plugins/resources/src/components/TemplatesPage/TemplatesPage.tsx [hunk 5]
```diff
@@ -1,13 +1,97 @@
 import React from 'react';
-import { Content, Page } from '@backstage/core';
+import {Content, Page, Table, TableColumn} from '@backstage/core';
+import {DummyResponse, Edge, Resource} from "../../../../../packages/zoo-api";
```
Comment (blfpd) [inline frontend/plugins/resources/src/components/TemplatesPage/TemplatesPage.tsx]:
```js
import {…} from "zoo-api";
```

**Review by blfpd — CHANGES_REQUESTED** [review]

File: frontend/plugins/resources/src/components/LibrariesPage/LibrariesPage.tsx (hunk 4 above)
Comment (gjuro87) [reply frontend/plugins/resources/src/components/LibrariesPage/LibrariesPage.tsx]: Fixed, i have used map as you suggested. It is cleaner

File: frontend/plugins/resources/src/components/LanguagesPage/LanguagesPage.tsx (hunk 3 above)
Comment (gjuro87) [reply frontend/plugins/resources/src/components/LanguagesPage/LanguagesPage.tsx]: fixed

File: frontend/plugins/resources/src/components/LanguagesPage/LanguagesPage.tsx (hunk 2 above)
Comment (gjuro87) [reply frontend/plugins/resources/src/components/LanguagesPage/LanguagesPage.tsx]: Used map()

File: frontend/plugins/resources/src/components/TemplatesPage/TemplatesPage.tsx (hunk 5 above)
Comment (gjuro87) [reply frontend/plugins/resources/src/components/TemplatesPage/TemplatesPage.tsx]: This was generated by my IDE, fixed

**Comment by gjuro87** [comment]
> The table search doesn’t pickup the name. I’m not sure how long would it be to fix it.
> Do you plan to add this feature in a later commit/MR?

@batisteo you are right. It doesn't work because i have used components like chip component as a value of table and search is doing compare by value. In case of compoment is object so it doesn’t match it. I will look at source code of backstage table and create our own table that matches our needs.

**Comment by gjuro87** [comment]
> > The table search doesn’t pickup the name. I’m not sure how long would it be to fix it.
> > Do you plan to add this feature in a later commit/MR?
> 
> @batisteo you are right. It doesn't work because i have used components like chip component as a value of table and search is doing compare by value. In case of compoment is object so it doesn’t match it. I will look at source code of backstage table and create our own table that matches our needs.

@batisteo fixed :)

**Comment by blfpd** [comment]
Great!

Could you address the DeepCode issues?

**Pull request merged**